tests/test_examples.py

```diff
@@ -86,6 +100,99 @@ def test_examples_files(test_file: str, symbols_by_file: Dict[str, Set[str]]) ->
+    """
+    Run python_ta on all the example files and return the map of file name to the
+    set of PythonTA messages it raises.
+
+    :param paths: The paths to retrieve the files from.
+    :param include_msg: whether to include msgs[msg] in the symbol set
+    """
+    sys.stdout = StringIO()
+    python_ta.check_all(
+        module_name=get_file_paths(paths),
+        config={
+            "output-format": "python_ta.reporters.JSONReporter",
+        }
+    )
+
+    jsons_output = sys.stdout.getvalue()
+    sys.stdout = sys.__stdout__
+    pyta_list_output = json.loads(jsons_output)
+
+    file_to_symbol = {}
+    for path, group in itertools.groupby(pyta_list_output, key=lambda d: os.path.basename(d["filename"])):
+        symbols = set()
+        for message in group:
+            for msg in message["msgs"]:
+                symbols.add(msg["symbol"])
+                if include_msg:
+                    symbols.add(msg["msg"])
+
+        file = os.path.basename(path)
+        file_to_symbol[file] = symbols
+
+    return file_to_symbol
+
+
+@pytest.fixture(scope="session")
+def check_examples_pyta() -> Dict[str, Set[str]]:
+    """
+    Checking the examples files with python_ta
+    """
+    return _symbols_by_file_pyta([_EXAMPLES_PATH, _CUSTOM_CHECKER_PATH])
+
+
+@pytest.fixture(scope="session")
+def check_pycodestyle_pyta() -> Dict[str, Set[str]]:
+    """
+    Checking the pycodestyle error files with python_ta, including the error message to
+    check for the specific error type
+    """
+    return _symbols_by_file_pyta([_PYCODESTYLE_PATH], include_msg=True)
+
+
+@pytest.mark.parametrize("test_file", get_file_paths([_EXAMPLES_PATH, _CUSTOM_CHECKER_PATH]))
+def test_examples_files_pyta(test_file: str, check_examples_pyta: Dict[str, Set[str]]) -> None:
+    """Creates all the new unit tests dynamically from the testing directory."""
+    base_name = os.path.basename(test_file)
+    if not re.match(_EXAMPLE_PREFIX_REGEX, base_name[:5]):
+        return
+    if not base_name.lower().endswith(".py"):
+        assert False
+    checker_name = base_name[6:-3].replace("_", "-")  # Take off prefix and file extension.
+
+    test_file_name = os.path.basename(test_file)
+    file_symbols = check_examples_pyta[test_file_name]
+
+    found_pylint_message = checker_name in file_symbols
+    assert found_pylint_message, f"Failed {test_file}. File does not add expected message."
+
+
+@pytest.mark.parametrize("test_file", get_file_paths(_PYCODESTYLE_PATH))
+def test_pycodestyle_errors_pyta(test_file: str, check_pycodestyle_pyta: Dict[str, Set[str]]) -> None:
+    """
+    Run python_ta on all pycodestyle error test cases
+    """
+    base_name = os.path.basename(test_file)
+    if not re.match(_PYCODESTYLE_PREFIX_REGEX, base_name[:4]):
+        return
+    if not base_name.lower().endswith(".py"):
+        assert False
+
+    # get the specific PEP8 error code
+    has_error = base_name[5:] == "error.py"
```

Comment:
Nice. This should really be a guard though (`if not has_error ... return`), or merged into the condition on line 178.